qemu/tests/cpu_device_hotplug_during_boot.py

```diff
@@ -70,10 +71,6 @@ def run(test, params, env):
 
         error_context.context("Check number of CPU inside guest after unplug.",
                               logging.info)
-        current_guest_cpus = vm.get_cpu_count()
-        if current_guest_cpus != smp:
-            test.fail("Actual number of guest CPUs(%s) is not equal to "
-                      "expected(%s) after hotunplug." % (current_guest_cpus,
-                                                         smp))
-        logging.info("CPU quantity(%s) in guest is correct.",
-                     current_guest_cpus)
+        if not cpu_utils.check_if_vm_vcpu_match(smp, vm):
```

Comment:
Suggest to not use smp directly. We might update cfg to add more cpu devices when testing sometimes, so prefer a more robust way.

Reply:
If you add more CPU devices and enable them in the cfg, the current test case will fail.

However, all CPU devices are hot plugged/unplugged during the boot stage, we cannot get the initial count in the guest at the beginning, only `smp` can be used if you don't want to use `vm.vcpu_threads`

Reply:
Can you explain why case would fail if add more cpu devices in cfg? 
I'm not asking to get initial count, but to calculate an expected cpu count by add smp and enabled cpu device.

Reply:
If there have some enabled vcpu device and some disabled vcpu devices in the cfg, the py will hotplug all of them in a loop, so those enabled vcpu devices will fail, because we have not a requirement of this scenario. Like append some vcpu devices in the qemu command line, and some idle vcpu devices, after launched the guest, hotplug idle vcpu devices + hotunplug those enabled vcpu devices.

Reply:
Sounds like a compatibility issue to me. Shouldn't only hotplug disabled vcpu devices? 
IMO, it's encouraged to cover more valid scenarios in automation.

Reply:
In our current test plan, all vcpu devices are plugged and unplugged in a loop, test fails if any device failed.
We missing a scenario like your opinion. So, if you require a flexible test case, I think we need a new polarion case to track it.